blocks/v2-hero/v2-hero.js

```diff
@@ -0,0 +1,98 @@
+import {
+  createElement,
+  isVideoLink,
+  removeEmptyTags,
+} from '../../scripts/scripts.js';
+
+const checkVideoTime = (event) => {
+  const video = event.target;
+  if (video.currentTime > 3) {
+    video.pause();
+    video.removeEventListener('timeupdate', checkVideoTime);
+  }
+};
+
+export default async function decorate(block) {
+  const blockName = 'v2-hero';
+
+  const picture = block.querySelector('picture');
+  const link = block.querySelector('a');
+  const isVideo = link ? isVideoLink(link) : false;
+  if (isVideo) {
+    const video = createElement('video', [`${blockName}__video`]);
+    video.muted = true;
+    video.autoplay = true;
+
+    const source = createElement('source', '', {
+      src: link.getAttribute('href'),
+      type: 'video/mp4',
+    });
+    video.appendChild(source);
+    block.prepend(video);
+    link.remove();
+
+    // prevent video to reproduce more than 3s
+    video.addEventListener('timeupdate', checkVideoTime);
+  }
+
+  if (picture) {
+    const img = picture.querySelector('img');
+    img.classList.add(`${blockName}__image`);
+    if (picture.parentElement.tagName === 'P') {
+      picture.parentElement.remove();
+    }
+    block.prepend(picture);
+  }
+
+  const contentWrapper = block.querySelector(':scope > div');
+  contentWrapper.classList.add(`${blockName}__content-wrapper`);
+
+  // add Hero text variants to the corresponding content
+  const variantClasses = ['centered', 'left', 'bottom', 'dark'];
+  variantClasses.forEach((variant) => {
+    if (block.classList.contains(variant)) {
+      if (variant !== 'dark') block.classList.remove(variant);
+      contentWrapper.classList.add(`${blockName}__content-wrapper--${variant}`);
```

Comment:
I think we should leave the classes with the block element. This way we have an easy way to see which variants are used by the block - using the element inspection. Additionally, we can modify any element in the block (for example, if we need to modify another element with the same variant in the future, we'll need just to add proper CSS).